### PR TITLE
docs: add the comment and version of dcurl API

### PR DESCRIPTION
**Pull request by marktwtn**
Add the comment and configuration file of Doxygen to generate the API document.
The README.md also adds a badge of the github release version.
At last, the new macros for dcurl version are added.

### CODE REVIEW

File: Doxyfile [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+#---------------------------------------------------------------------------
+QUIET                  = YES
+#---------------------------------------------------------------------------
+# Configuration options related to the input files
+#---------------------------------------------------------------------------
+INPUT                  = . \
+                         docs \
+                         src
+FILE_PATTERNS          = dcurl.c \
+                         dcurl.h \
```
Comment (wusyong) [inline Doxyfile:19]: I think documentation could just all put in dcurl.h

Comment (marktwtn) [reply Doxyfile:19]: I searched on the Internet and found out that writing the document in C source file or header file is acceptable.
It only suggests that we should not write the document on both side.

The reason I wrote in source file is if the functions are changed, they can be modified without opening another file.

I think no matter which way we choose, just make sure every project follows the same rule.

**Review by wusyong — APPROVED** [review]

File: src/dcurl.c [hunk 2]
```diff
@@ -99,6 +117,17 @@ void dcurl_destroy()
...
 
 
+/**
+ * @brief dcurl execution.
+ *
+ * Retrieve the available hardware from the list and use it to do the
+ * PoW(Proof-of-Work).
+ * @param [in] trytes The trytes for PoW calculation.
+ * @param [in] mwm The minimum weight magnitude.
+ * @param [in] threads The thread number of calculating the PoW. It affects CPU
```
Comment (wusyong) [inline src/dcurl.c]: Detailed explanation of thread amount may needed (like 0 is actually max amount - 1.)

File: Doxyfile (hunk 1 above)
Comment (marktwtn) [reply Doxyfile:19]: The comment has been moved to the header file.

File: src/dcurl.c (hunk 2 above)
Comment (marktwtn) [reply src/dcurl.c]: The explanation of threads has been updated.

File: Doxyfile (hunk 1 above)
Comment (marktwtn) [reply Doxyfile:19]: I did not remove the `dcurl.c` in `FILE_PATTERNS` since I want to make the users to link and check the source code easily.

File: src/dcurl.c [hunk 3]
```diff
@@ -9,11 +9,10 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <unistd.h>
-#if defined(ENABLE_OPENCL)
-#include "pow_cl.h"
-#endif
-#if defined(ENABLE_FPGA_ACCEL)
-#include "pow_fpga_accel.h"
+#ifdef __APPLE__
```
Comment (jserv) [inline src/dcurl.c]: Why did you change `src/dcurl.c`?

Comment (marktwtn) [reply src/dcurl.c]: The original `Doxygen` comment was written in the `dcurl.c`.
I also changed the order of the header files for better classification.

Since the comment has been moved to `dcurl.h`, I just left the rest of the modification in the commit.

Comment (marktwtn) [reply src/dcurl.c]: Shall I revert `dcurl.c` for a cleaner commit?

Comment (jserv) [reply src/dcurl.c]: Yes, please do. Only preserve functional changes.

File: src/dcurl.h [hunk 4]
```diff
@@ -4,8 +4,44 @@
 #include <stdbool.h>
 #include "trinary.h"
 
+/**
+ * @file dcurl.h
+ */
```
Comment (jserv) [inline src/dcurl.h:16]: Add general descriptions about dcurl here.

File: src/dcurl.c (hunk 3 above)
Comment (marktwtn) [reply src/dcurl.c]: Done.

File: src/dcurl.h (hunk 4 above)
Comment (marktwtn) [reply src/dcurl.h:16]: Done.

Comment (marktwtn) [reply src/dcurl.h:16]: Did you mean the general descriptions about dcurl rather than the description of this file?
I think I might update the comment with the latter one.

Comment (jserv) [reply src/dcurl.h:16]: It would be something describing the functionality about dcurl.
> dcurl library is provided as hardware-accelerated implementation for IOTA PearlDiver, which utilizes multi-threaded SIMD, FPGA and GPU.

Comment (marktwtn) [reply src/dcurl.h:16]: Done.
